Microsoft.Toolkit.Uwp.SampleApp/SamplePages/CameraPreview/CameraPreviewPage.xaml.cs

```diff
@@ -0,0 +1,147 @@
+    /// </summary>
+    public sealed partial class CameraPreviewPage : Page, IXamlRenderListener
+    {
+        private static SemaphoreSlim semaphoreSlim;
+        private VideoFrame _currentVideoFrame;
+        private SoftwareBitmapSource _softwareBitmapSource;
+        private CameraPreview _cameraPreviewControl;
+        private Image _imageControl;
+        private TextBlock _errorMessageText;
+
+        public CameraPreviewPage()
+        {
+            this.InitializeComponent();
+            semaphoreSlim = new SemaphoreSlim(1);
+        }
+
+        public async void OnXamlRendered(FrameworkElement control)
+        {
+            // Using a semaphore lock for synchronocity. 
+            // This method gets called multiple times when accessing the page from Latest Pages
+            // and creates unused duplicate references to Camera and memory leaks.
+            await semaphoreSlim.WaitAsync();
+
+            await CleanUpAsync();
+            _cameraPreviewControl = control.FindDescendantByName("CameraPreviewControl") as CameraPreview;
+
+            if (_cameraPreviewControl != null)
+            {
+                var cameraHelper = new CameraHelper();
+                await _cameraPreviewControl.SetCameraHelperAsync(cameraHelper);
+                _cameraPreviewControl.CameraHelper.FrameArrived += CameraPreviewControl_FrameArrived;
+                _cameraPreviewControl.PreviewFailed += CameraPreviewControl_PreviewFailed;
+            }
+
+            _imageControl = control.FindDescendantByName("CurrentFrameImage") as Image;
+            if (_imageControl != null)
+            {
+                _softwareBitmapSource = new SoftwareBitmapSource();
+                _imageControl.Source = _softwareBitmapSource;
+            }
+
+            _errorMessageText = control.FindDescendantByName("ErrorMessage") as TextBlock;
+
+            semaphoreSlim.Release();
+        }
+
+        protected override void OnNavigatedTo(NavigationEventArgs e)
+        {
+            base.OnNavigatedTo(e);
+            Shell.Current.RegisterNewCommand("Capture Current Frame", CaptureButton_Click);
+            Application.Current.Suspending += Application_Suspending;
+            Application.Current.Resuming += Application_Resuming;
+        }
+
+        protected override void OnNavigatedFrom(NavigationEventArgs e)
+        {
+            CleanUpAsync();
+        }
+
+        private async void Application_Suspending(object sender, SuspendingEventArgs e)
+        {
+            if (Frame.CurrentSourcePageType == typeof(CameraPreviewPage))
+            {
+                var deferral = e.SuspendingOperation.GetDeferral();
+                await CleanUpAsync();
+                deferral.Complete();
+            }
+        }
+
+        private async void Application_Resuming(object sender, object e)
+        {
+            var cameraHelper = new CameraHelper();
+            await _cameraPreviewControl.SetCameraHelperAsync(cameraHelper);
```

Comment:
The CameraHelper should react to these events instead of the user having to do it. When the app suspends, the CameraHelper should automatically cleanup itself, raise an event to let the CameraPreview control know so it stops the preview. When the app resumes, the CameraHelper should build up itself and notify the CameraPreview control so it starts the preview. This makes the helper and control much more dev friendly.